libraries/lib-wave-track/WaveClip.h

```diff
@@ -257,7 +258,10 @@ class WAVE_TRACK_API WaveClip final : public XMLTagHandler
    /*!
     @pre `ii < GetWidth()`
     */
-   Sequence* GetSequence(size_t ii) { return mSequences[ii].get(); }
+   Sequence* GetSequence(size_t ii) {
+      assert(ii < GetWidth());
+      return mSequences[ii].get();
```

Comment:
I thought that most of `std::vector` implementations do perform out-of-range check in debug. MS does for sure.

Reply:
Maybe so, but this would fail a little faster.

if I state a precondition in a way formal enough to be coded, and the test is not expensive at run time, then I also assert it in the function body for completeness.

I think assertions properly used are not only for detecting bugs.  They are also for documenting assumptions and for assisting the reasoning of someone reading the code.